node/src/services/mod.rs

```diff
@@ -1,34 +1,61 @@
 pub mod blockchain;
 pub mod file_transfer;
 
+use std::sync::Arc;
+
 use storage_hub_infra::{
     actor::{ActorHandle, TaskSpawner},
     event_bus::EventHandler,
+    storage::{FileStorage, ForestStorage},
 };
+use tokio::sync::RwLock;
 
 use crate::tasks::{
     AcceptedBspVolunteerHandler, NewStorageRequestHandler, ResolveRemoteUploadRequest,
 };
 
 use self::{blockchain::handler::BlockchainService, file_transfer::FileTransferService};
 
-#[derive(Debug, Clone)]
-pub struct StorageHubHandler {
+pub trait StorageHubHandlerConfig: Send + 'static {
```

Comment:
```suggestion
pub trait StorageHubHandlerConfig: Send + 'static + Clone {
```

Reply:
By adding this constraint, you can use `#[derive(Clone)]` in the task definitions like `ResolveRemoteUploadRequest`, and not have to implement `Clone` manually. Less boilerplate code, and I believe it is safe to ask this constraint from `StorageHubHandlerConfig`.

Reply:
If we add `Clone` bound to `StorageHubHandlerConfig`, it will pass the constraint to `FileStorage` and `ForestStorage`. We can change it like this but we have to make sure we never actually `clone` those types, which imo is worse thank manually implementing this.
Also the boilerplate code can be fixed later (with our own macro).